client-src/elements/chromedash-feature-page.ts

```diff
@@ -164,13 +184,74 @@ export class ChromedashFeaturePage extends LitElement {
           this.closestShippingDate = channels[key].final_beta;
         }
       });
+    } else {
+      // If not upcoming, find the closest milestone that has shipped.
+      let latestMilestone = 0;
+      for (const ms of shippingMilestones) {
+        if (ms && ms <= latestStableVersion) {
+          latestMilestone = Math.max(latestMilestone, ms);
+        }
+      }
+
+      if (latestMilestone === latestStableVersion) {
+        this.closestShippingDate = channels['stable']?.final_beta;
+        this.hasShipped = true;
+      } else {
+        this.closestShippingDate =
+          await this.fetchClosestShippingDate(latestMilestone);
+        this.hasShipped = true;
+      }
+    }
+  }
+
+  /**
+   * Determine if it should show warnings to a feature author, if
+   * a shipped feature is outdated, and it has edit access.*/
+  isShippedFeatureOutdatedForAuthor() {
+    return this.userCanEdit() && this.isShippedFeatureOutdated();
+  }
+
+  /**
+   * Determine if it should show warnings to all readers, if
+   * a shipped feature is outdated, and last update was > 2 months.*/
+  isShippedFeatureOutdatedForAll() {
+    if (!this.isShippedFeatureOutdated()) {
+      return false;
     }
+
+    // Represent two months grace period.
+    const nineWeekPeriod = 9 * 7 * 24 * 60 * 60 * 1000;
+    const isVerified = isVerifiedWithinGracePeriod(
+      this.feature.accurate_as_of,
+      this.currentDate,
+      nineWeekPeriod
+    );
+    return !isVerified;
+  }
+
+  /**
+   * A feature is outdated if it has shipped, and its
+   * accurate_as_of is before its latest shipping date before today.*/
+  isShippedFeatureOutdated(): boolean {
+    // Check if a feature has shipped.
+    if (!this.hasShipped) {
+      return false;
+    }
+
+    if (!this.feature.accurate_as_of) {
```

Comment:
Should this case return false?

We set accurate_as_of when a feature is created.  The code to do that has been live for about two years.  Anything that still lacks accurate_as_of is probably an old entry.

If this returns true, then lots of old entries will start to show this banner.  I doubt that the feature owners for those old features will come back to verify and set accurate_as_of.

Reply:
I see. That is a fair point! I will treat it as not outdated and will leave a comment inline